python/ray/exceptions.py

```diff
@@ -292,19 +295,103 @@ def __init__(self, object_ref_hex, call_site):
             ray_constants.CALL_STACK_LINE_DELIMITER, "\n  ")
 
     def __str__(self):
-        msg = (f"Object {self.object_ref_hex} cannot be retrieved due to node "
-               "failure or system error.")
+        msg = f"Object {self.object_ref_hex} cannot be retrieved. "
         if self.call_site:
-            msg += (f" The ObjectRef was created at: {self.call_site}")
+            msg += (f"The ObjectRef was created at: {self.call_site}")
         else:
             msg += (
-                " To see information about where this ObjectRef was created "
+                "To see information about where this ObjectRef was created "
                 "in Python, set the environment variable "
                 "RAY_record_ref_creation_sites=1 during `ray start` and "
```

Comment:
We also need to set this for driver. Why don't we write a doc and make a link instead? (I think it is confusing for users anyway when they just read it)

Reply:
I added a doc, but left it out of the error message for now. I think it's best if we try to make the error messages standalone.